source/dal/objects/disk.py

```diff
@@ -18,16 +18,99 @@
 This is the Disk's module
 """
 
+import os
 from source.dal.base import Base
+from source.tools.localclient import LocalClient
 
 
 class Disk(Base):
     """
     Represents a disk on the system.
     """
 
+    _local_client = LocalClient()
+
     _table = 'disk'
-    _properties = [['name', int],
+    _properties = [['name', str],
+                   ['state', str],
                    ['aliases', list],
-                   ['mountpoint', str]]
+                   ['is_ssd', bool],
+                   ['model', str],
+                   ['size', int],
+                   ['serial', str],
+                   ['partitions', dict]]
     _relations = []
+    _dynamics = ['mountpoint', 'available', 'usable', 'status', 'usage', 'partition_aliases']
+
+    def _mountpoint(self):
+        for partition in self.partitions:
+            mountpoint = partition['mountpoint']
+            if mountpoint is not None:
+                return mountpoint
+        return None
+
+    def _available(self):
+        return self.mountpoint is None or not self.mountpoint.startswith('/mnt/alba-asd/')
+
+    def _usable(self):
+        mountpoints = []
+        for partition in self.partitions:
+            mountpoint = partition['mountpoint']
+            if mountpoint is not None:
+                mountpoints.append(mountpoint)
+        if len(mountpoints) > 1:
+            return False  # Multiple mountpoints: Not supported
+        if self.mountpoint is not None:
+            # Only one mountpoint. Accept if it managed by us
+            if not self.mountpoint.startswith('/mnt/alba-asd/'):
+                return False
+            return True
+        # No mountpoint(s): Search for "forbidden" partition types
+        for partition in self.partitions:
+            partition_filesystem = partition['filesystem']
+            if partition_filesystem in ['swap', 'linux_raid_member', 'LVM2_member']:
+                return False
+        return True
+
+    def _status(self):
+        if self.mountpoint is not None:
+            if self.state == 'MISSING':
+                return {'state': 'error',
+                        'detail': 'missing'}
+            output, error = self._local_client.run(['ls', '{0}/'.format(self.mountpoint)],
+                                                   allow_nonzero=True, return_stderr=True, timeout=5)
+            output += error
+            if 'Input/output error' in output:
+                return {'state': 'error',
+                        'detail': 'io_error'}
+        return {'state': 'ok'}
+
+    def _usage(self):
+        if self.mountpoint is not None:
+            df_info = self._local_client.run("df -B 1 --output=size,used,avail '{0}' | tail -1 || true".format(self.mountpoint.replace(r"'", r"'\''")),
```

Comment:
`allow_nonzero` should be True. If the mountpoint is corrupt, this might raise otherwise